### PR TITLE
feat(state-importer): integrate state importer to CLI

### LINKED ISSUE #77: Integrate State Importer
### Description

Integrate the [new state importer](https://github.com/commercetools/nodejs/issues/433) module

### PR DESCRIPTION
**Pull request by wizzy25**
#### Summary

Integrate the state import module
#### Description

- This PR adds [this module](https://github.com/commercetools/nodejs/pull/504) to the import CLI
- Drops support for Node v4
- Updates productType import dependency, which now requires productType keys
- Bumps dependencies

resolves #77

### CODE REVIEW

File: features/support/hooks.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { Before, After } from 'cucumber'
```
Comment (junajan) [inline features/support/hooks.js]: Is this file needed?

File: features/step_definitions/cli_steps.js [hunk 2]
```diff
@@ -6,145 +6,149 @@ import rimraf from 'rimraf'
...
-            cleansingNeeded = true
-            callback()
-          })
-        })
+Given(
+  /^a file named "(.*)" with:$/,
+  (filePath, fileContent, callback) => {
+    cleanseIfNeeded()
+    const absoluteFilePath = `${tmpDir}/${filePath}`
+    const filePathSegments = absoluteFilePath.split('/')
```
Comment (junajan) [inline features/step_definitions/cli_steps.js]: We can use `path` module which will handle also a difference between path separator on Windows vs *nix systems.

File: features/step_definitions/cli_steps.js [hunk 3]
```diff
@@ -6,145 +6,149 @@ import rimraf from 'rimraf'
...
+    filePathSegments.pop()
+    const dirName = joinPathSegments(filePathSegments)
```
Comment (junajan) [inline features/step_definitions/cli_steps.js]: same as above - `path.dirname(absoluteFilePath)`

File: features/step_definitions/cli_steps.js [hunk 4]
```diff
@@ -6,145 +6,149 @@ import rimraf from 'rimraf'
...
+          process.chdir(initialCwd)
+          cleansingNeeded = true
+          callback()
+        })
+      })
+    })
+  },
+)
+
+Then(/^the exit status should be ([0-9]+)$/, (code, callback) => {
```
Comment (junajan) [inline features/step_definitions/cli_steps.js]: I thought that cucumber should already know these "commands", why we define them here?

File: features/step_definitions/cli_steps.js [hunk 5]
```diff
@@ -6,145 +6,149 @@ import rimraf from 'rimraf'
...
+    loadCredentials().then((credentials) => {
+      const oauth2 = new OAuth2({ config: credentials })
+      oauth2.getAccessToken((err, response, body) => {
+        if (err)
+          throw new Error(`Error occurred when fetching accessToken ${err}`)
+        const accessToken = body.access_token
+        const initialCwd = process.cwd()
+        process.chdir(tmpDir)
+
+        const runtimePath = joinPathSegments([ baseDir, 'bin', 'sphere' ])
```
Comment (junajan) [inline features/step_definitions/cli_steps.js]: `path.join`

File: test/commands/import.spec.js [hunk 6]
```diff
@@ -3,6 +3,8 @@ import test from 'tape'
 import ImportCommand from '../../lib/commands/import'
 
 const _commander = require('rewire')('commander')
+// Increase max listeners as new ones are added for wach test case
```
[the remark this answers is not in the document]
Comment (wizzy25) [reply test/commands/import.spec.js]: `each` ;-)

File: test/commands/import.spec.js [hunk 7]
```diff
@@ -176,25 +182,27 @@ test(`ImportCommand
   should process price command`, (t) => {
   before().then((command) => {
     const spy1 = sinon.stub(command, '_stream')
-    const spy2 = sinon.stub(command, '_preProcess', opts => command._process(
-        Object.assign(opts, { config: {}, credentials: fakeCredentials }))
-    )
+    const spy2 = sinon.stub(command, '_preProcess').callsFake(opts =>
```
[the remark this answers is not in the document]
Comment (wizzy25) [reply test/commands/import.spec.js]: I didn't rewrite the whole file in general, just did a bit of a refactor. Will nevertheless modify this

File: features/support/hooks.js (hunk 1 above)
Comment (wizzy25) [reply features/support/hooks.js]: Yes, and it should setup/teardown the environment. This should be done in another PR though

File: features/step_definitions/cli_steps.js (hunk 4 above)
Comment (junajan) [reply features/step_definitions/cli_steps.js]: For example here https://github.com/sphereio/sphere-product-type-json-generator/blob/master/features/common.feature#L5 we do not define what does "Then the exit status should be 1" mean and it works..

Comment (wizzy25) [reply features/step_definitions/cli_steps.js]: Yeah. I think the more assertions the better. And as the test passes, I don't see a reason to remove it
